### PR TITLE
add pre-commit to gitlab

### PR DESCRIPTION
**Pull request by ilaner**
Adds pre-commit to gitlab ci, including content and sdk nightly.

fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9486?filter=-1
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9591?filter=-1
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-7058?filter=-1
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9475?filter=-1

content nightly: https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/910881
content nightly passed, only abnormal modeling rules failed and it fails in the regular nightly as well

### CODE REVIEW

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 1]
```diff
@@ -497,6 +498,28 @@
     matrix:
       - DOCKER: ['native:ga,native:maintenance,native:candidate','native:dev,from-yml']
 
+.run-pre-commit:
+  extends:
+    - .unittests-and-lint-settings
+  script:
+    - section_start "Revoking GCP Auth and Configure Docker"
```
Comment (dorschw) [inline .gitlab/ci/.gitlab-ci.global.yml:505]: add a comment why

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 2]
```diff
@@ -99,6 +99,26 @@ run-unittests-and-lint:
...
     - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
 
+run-pre-commit:
+  extends:
+    - .run-pre-commit
+  cache:
+    policy: pull-push
+
+  rules:
+    - if: '$BUCKET_UPLOAD == "true"'
```
Comment (dorschw) [inline .gitlab/ci/.gitlab-ci.on-push.yml:109]: needed?

Comment (dorschw) [inline .gitlab/ci/.gitlab-ci.on-push.yml:109]: open an issue for the bucket pre-commit please

File: .gitlab/ci/.gitlab-ci.on-push.yml [hunk 3]
```diff
@@ -99,6 +99,26 @@ run-unittests-and-lint:
...
+      when: never
+    - if: '$SECURITY_SCANS == "true"'
+      when: never
+    - if: '$BUILD_MACHINES_CLEANUP == "true"'
+      when: never
+    - if: '$FORCE_BUCKET_UPLOAD == "true"'
+      when: never
+    - if: '$DEMISTO_TEST_NATIVE_CANDIDATE == "true"'
+      when: never
+    - if: '$DEMISTO_SDK_NIGHTLY != "true"'
```
Comment (dorschw) [inline .gitlab/ci/.gitlab-ci.on-push.yml:119]: please comment why

File: .pre-commit-config_template.yaml [hunk 4]
```diff
@@ -98,7 +133,7 @@ repos:
     entry: python
     language: docker
     files: Packs\/.*_test\.py$
-    exclude: test_data
+    exclude: test_data|Packs/ApiModules/Scripts/NGINXApiModule/NGINXApiModule_test.py|Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py|Packs/Base/Scripts/SanePdfReport/SanePdfReport_test.py|Packs/Base/Scripts/CheckDockerImageAvailable/CheckDockerImageAvailable_test.py|Packs/CommonScripts/Scripts/Ping/Ping_test.py|Packs/CommonScripts/Scripts/GetDockerImageLatestTag/GetDockerImageLatestTag_test.py|Packs/CommonScripts/Scripts/GetDuplicatesMlv2/GetDuplicatesMlv2_test.py|Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck_test.py|Packs/rasterize/Integrations/rasterize/rasterize_test.py|Packs/Slack/Integrations/Slack/Slack_test.py|Packs/SymantecDLP/Integrations/SymantecDLP/SymantecDLP_test.py|Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository_test.py|Packs/Whois/Integrations/Whois/Whois_test.py
```
Comment (dorschw) [inline .pre-commit-config_template.yaml:136]: add comment why

File: Tests/scripts/dev_envs/pytest/conftest.py [hunk 5]
```diff
@@ -53,6 +53,13 @@ def test_boo(capfd):
         pytest.fail("Found output in stderr: [{}]".format(err.strip()))
 
 
+def pytest_sessionfinish(session, exitstatus):
+    """
```
Comment (dorschw) [inline Tests/scripts/dev_envs/pytest/conftest.py:57]: document please

File: pyproject.toml [hunk 6]
```diff
@@ -491,6 +491,7 @@ src = [
 "**/*_test.py" = []
 # Ignore any Python update in ApiModules, until we can proprtly determine their desired version.
 "**/*ApiModule.py"  = [ "UP" ]
+"conftest.py" = [ "UP" ]
```
Comment (dorschw) [inline pyproject.toml:494]:
```suggestion

```

File: .pre-commit-config_template.yaml (hunk 4 above)
Comment (ilaner) [reply .pre-commit-config_template.yaml:118]: added

File: Tests/scripts/dev_envs/pytest/conftest.py (hunk 5 above)
Comment (ilaner) [reply Tests/scripts/dev_envs/pytest/conftest.py:57]: documented

File: .gitlab/ci/.gitlab-ci.on-push.yml (hunk 3 above)
Comment (ilaner) [reply .gitlab/ci/.gitlab-ci.on-push.yml:121]: commented

**Review by dorschw — APPROVED** [review]

**Pull request merged**